src/librustc/hir/map/def_collector.rs

```diff
@@ -27,7 +27,7 @@ pub struct DefCollector<'a> {
     hir_crate: Option<&'a hir::Crate>,
     definitions: &'a mut Definitions,
     parent_def: Option<DefIndex>,
-    pub visit_macro_invoc: Option<&'a mut FnMut(NodeId, DefIndex)>,
+    pub visit_macro_invoc: Option<&'a mut FnMut(NodeId, DefIndex, bool /* const_integer */)>,
```

Comment:
I could refactor the closure into a user-defined trait object (e.g. `Option<&'a mut MacroInvocationVisitor>`, where `MacroInvocationVisitor` is a trait with method `fn visit_invoc(id: NodeId, def_index: DefIndex, const_integer: bool)`).

The invocation visiting code needs to be in `resolve`, so we need a trait object of some sort.